tests/integration/orcid/test_orcid_import_legacy_tokens.py

```diff
@@ -161,6 +163,40 @@ def test_multiple_list_generator(app, redis_setup):
     assert redis_setup.llen('legacy_orcid_tokens') == 0
 
 
+class ORCIDLegacyTokensConsumer(threading.Thread):
+    def __init__(self, app):
+        threading.Thread.__init__(self)
+        self.app = app
+        self.tokens = []
+
+    def run(self):
+        with self.app.app_context():
+            self.tokens = list(legacy_orcid_arrays())
+
+
+def test_legacy_orcid_arrays_concurrency(app, redis_setup):
+    """
+    Ensure legacy_orcid_arrays() works with concurrent consumers.
```

Comment:
I am not sure this test is asserting that because of the GIL. Besides we don't usually put docstrings in tests, as the function name should be self-explanatory.

Reply:
Done. You are right about threads and GIL, thus as we discussed I removed this test.